optas/models.py

```diff
@@ -217,6 +263,18 @@ def upper_actuated_joint_limits(self):
     def velocity_actuated_joint_limits(self):
         return cs.DM([jnt.limit.velocity for jnt in self._urdf.joints if jnt.type != 'fixed'])
 
+    @property
+    def lower_opt_joint_limits(self):
+        return cs.DM([jnt.limit.lower for jnt in self._urdf.joints if jnt.name in self.opt_joint_names])
+
+    @property
+    def upper_opt_joint_limits(self):
+        return cs.DM([jnt.limit.upper for jnt in self._urdf.joints if jnt.name in self.opt_joint_names])
+
+    @property
+    def velocity_opt_joint_limits(self):
```

Comment:
Perhaps we should introduce acceleration limits also. Perhaps instead of multiple properties for each upper/lower limits we should have one method that returns the limits given a `time_deriv`.

Reply:
Yes, but i would leave this for another issue/pull request. Otherwise it is already a lot we are changing here.